core/src/tx_sender.rs

```diff
@@ -162,20 +173,23 @@ impl TxSender {
     /// Tries to send a tx. If all conditions are met, it will save the tx to the database.
     /// It will also save the cancelled outpoints, cancelled txids and activated prerequisite txs to the database.
     /// It will automatically saves inputs as cancelled outpoints.
+    /// It will automatically saves inputs as activated outpoints.
     #[tracing::instrument(err(level = tracing::Level::ERROR), ret(level = tracing::Level::TRACE))]
     pub async fn try_to_send(
         &self,
         dbtx: DatabaseTransaction<'_, '_>,
+        tx_data_for_logging: Option<TxDataForLogging>,
         signed_tx: &Transaction,
         fee_paying_type: FeePayingType,
         cancel_outpoints: &[OutPoint],
         cancel_txids: &[Txid],
-        activate_txids: &[ActivedWithTxid],
-        activate_outpoints: &[ActivedWithOutpoint],
+        activate_txids: &[ActivatedWithTxid],
+        activate_outpoints: &[ActivatedWithOutpoint],
+        skip_saving_inputs_as_activated: bool,
```

Comment:
way too many args. At this point, it's better to have a struct for this with a builder.

You can use the `derive_builder` crate for easy derivation.